editor/src/messages/layout/layout_message_handler.rs

```diff
@@ -48,6 +48,89 @@ impl<F: Fn(&MessageDiscriminant) -> Vec<KeysGroup>> MessageHandler<LayoutMessage
 		use LayoutMessage::*;
 		#[remain::sorted]
 		match message {
+			CommitLayout { layout_target, widget_id, value } => {
+				let layout = if let Some(layout) = self.layouts.get_mut(layout_target as usize) {
+					layout
+				} else {
+					warn!("CommitLayout was called referencing an invalid layout. `widget_id: {widget_id}`, `layout_target: {layout_target:?}`",);
+					return;
+				};
+
+				let widget_holder = if let Some(widget_holder) = layout.iter_mut().find(|widget| widget.widget_id == widget_id) {
+					widget_holder
+				} else {
+					warn!("CommitLayout was called referencing an invalid widget ID, although the layout target was valid. `widget_id: {widget_id}`, `layout_target: {layout_target:?}`",);
+					return;
+				};
+
+				#[remain::sorted]
+				match &mut widget_holder.widget {
```

Comment:
I'm not a big fan of having this match statement for all the widget types exist twice (also on lines 163-291). Would you mind breaking it out into a function and combining the two use cases into one so it does the update or commit logic depending on the function's arguments?